src/libData/BlockData/BlockHeader/VCBlockHeader.cpp

```diff
@@ -0,0 +1,228 @@
+* another public or private blockchain network. This source code is provided ‘as is’ and no 
+* warranties are given as to title or non-infringement, merchantability or fitness for purpose 
+* and, to the extent permitted by law, all liability for your use of the code is disclaimed. 
+* Some programs in this code are governed by the GNU General Public License v3.0 (available at 
+* https://www.gnu.org/licenses/gpl-3.0.en.html) (‘GPLv3’). The programs that are governed by 
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include "VCBlockHeader.h"
+#include "libUtils/Logger.h"
+
+using namespace std;
+using namespace boost::multiprecision;
+
+VCBlockHeader::VCBlockHeader() { m_CandidateLeaderIndex = 1; }
+
+VCBlockHeader::VCBlockHeader(const vector<unsigned char>& src,
+                             unsigned int offset)
+{
+    if (Deserialize(src, offset) != 0)
+    {
+        LOG_GENERAL(INFO, "Error. We failed to initialize VCBlockHeader.");
+    }
+}
+
+VCBlockHeader::VCBlockHeader(const uint64_t& vieWChangeDSEpochNo,
+                             const uint64_t& viewChangeEpochNo,
+                             const unsigned char viewChangeState,
+                             const unsigned int expectedCandidateLeaderIndex,
+                             const Peer& candidateLeaderNetworkInfo,
+                             const PubKey& candidateLeaderPubKey,
+                             const unsigned int vcCounter,
+                             const boost::multiprecision::uint256_t& timestamp)
+    : m_VieWChangeDSEpochNo(vieWChangeDSEpochNo)
+    , m_VieWChangeEpochNo(viewChangeEpochNo)
+    , m_ViewChangeState(viewChangeState)
+    , m_CandidateLeaderIndex(expectedCandidateLeaderIndex)
+    , m_CandidateLeaderNetworkInfo(candidateLeaderNetworkInfo)
+    , m_CandidateLeaderPubKey(candidateLeaderPubKey)
+    , m_VCCounter(vcCounter)
+    , m_Timestamp(timestamp)
+{
+}
+
+unsigned int VCBlockHeader::Serialize(vector<unsigned char>& dst,
+                                      unsigned int offset) const
+{
+    LOG_MARKER();
+
+    unsigned int size_needed = VCBlockHeader::SIZE;
+    unsigned int size_remaining = dst.size() - offset;
+
+    if (size_remaining < size_needed)
+    {
+        dst.resize(size_needed + offset);
+    }
+
+    unsigned int curOffset = offset;
+    SetNumber<uint64_t>(dst, curOffset, m_VieWChangeDSEpochNo,
+                        sizeof(uint64_t));
+    curOffset += sizeof(uint64_t);
+    SetNumber<uint64_t>(dst, curOffset, m_VieWChangeEpochNo, sizeof(uint64_t));
+    curOffset += sizeof(uint64_t);
+    SetNumber<unsigned char>(dst, curOffset, m_ViewChangeState,
+                             sizeof(unsigned char));
+    curOffset += sizeof(unsigned char);
+    SetNumber<unsigned int>(dst, curOffset, m_CandidateLeaderIndex,
```

Comment:
Change sizeof(unsigned char) to sizeof(uint8_t). We should keep the length of a field to be fixed, as we cannot predict which os is used by another node.